python/ray/serve/autoscaling_policy.py

```diff
@@ -88,16 +89,36 @@ def _calculate_desired_num_replicas(
     return desired_num_replicas
 
 
+@dataclass
 @PublicAPI(stability="alpha")
-def replica_queue_length_autoscaling_policy(
-    curr_target_num_replicas: int,
-    current_num_ongoing_requests: List[float],
-    current_handle_queued_queries: float,
-    config: Optional[AutoscalingConfig],
-    capacity_adjusted_min_replicas: int,
-    capacity_adjusted_max_replicas: int,
-    policy_state: Dict[str, Any],
-) -> int:
+class AutoscalingContext:
+    """The context for an autoscaling policy call."""
+
+    # The AutoscalingConfig which the deployment started with.
+    config: AutoscalingConfig
```

Comment:
I think this should just be a dictionary right? Other policies will have their own custom config fields

Reply:
Right now in the design doesn't allow custom configs on `AutoscalingConfig` to pass in any config fields. This `config` is just passing the existing `AutoscalingConfig`. We can cast it to dictionary, but I'm not sure how useful it may be. 

I think it shouldn't be too bad to add any custom configs tho, just need to do some kinda of `setattr` on the `AutoscalingConfig` object and make sure the protobuf serdes continue to work. Can be a follow up if you think this is needed.